src/mme-app/handlers/attach_initue.c

```diff
@@ -162,10 +164,13 @@ stage1_processing(struct s6a_Q_msg *s6a_req, struct attachReqRej_info *s1ap_rej,
       unsigned char mnc1 = plmn_byte2 >> 4;  // mnc1
       unsigned char mcc3  = plmn_byte2 & 0xf; //mcc3
       // First byte we are not changing             mcc2 mcc1
-      plmn_byte2 = (mnc3 << 4) | mcc3; // 2nd byte on NAS - mnc3 mcc3
-      plmn_byte3 = (mnc2 << 4) | mnc1; // 3rd byte on NAS - <mnc2 mnc1>
-      ue_info->tai.plmn_id.idx[1] = plmn_byte2;
-      ue_info->tai.plmn_id.idx[2] = plmn_byte3;
+      if(mnc1 != 0x0F)
```

Comment:
I will review this separately. Requires more time to see if we are aligned with all 4 encoder/decoder - s11, s6a, s1ap & NAS.